### DISCUSSION TITLE
Creating own ISO images

### DISCUSSION THREAD

**Discussion by romangg** [r0b0/debian-installer#24] [Q&A]
I want to test some functionality I'm working on by creating a full disk image like described [here](https://github.com/romangg/debian-installer/tree/master#creating-your-own-installer-image).

I'm not sure how to apply this guide when I only want to create a .iso file that I can then run in a VM. Is it doable at all?

Also I noticed the images that are provided from the readme are .img files. Is there a tangible difference? I remember the last time I tried on a server I don't have access to right now Proxmox had problems with recognizing the .img extension when loading for installing to a VM.

**Comment by r0b0** [comment]
Usually, iso images are usually CD images, I think the name comes from the iso 9660 standard https://en.m.wikipedia.org/wiki/ISO_9660

The opinionated installer images are not CD images but rather read-write disk images and they are not iso9660 but GPT.

**Reply by romangg** [reply]
Ah ok. Thanks for the clarification. So is it possible to directly write into a "virtual image" i.e. a file instead of a "blank storage device" i.e. a usb stick? Or do I misunderstand it on a fundamental level?

**Comment by r0b0** [comment]
Of courses it can be a virtual image. That's how I make the "official" images. I just attach an image file to a VM.
